Error on compilation:

✘ [ERROR] TS2339: Property 'toggleNav' does not exist on type 'App'. [plugin angular-compiler]

    src/app/app.html:6:12:
      6 │       (click)="toggleNav()"
        ╵                ~~~~~~~~~

  Error occurs in the template of component App.

    src/app/app.ts:8:15:
      8 │   templateUrl: './app.html',
        ╵                ~~~~~~~~~~~~


✘ [ERROR] TS2339: Property 'isNavOpen' does not exist on type 'App'. [plugin angular-compiler]

    src/app/app.html:7:25:
      7 │       [attr.aria-expanded]="isNavOpen"
        ╵                             ~~~~~~~~~

  Error occurs in the template of component App.

    src/app/app.ts:8:15:
      8 │   templateUrl: './app.html',
        ╵                ~~~~~~~~~~~~


✘ [ERROR] TS2339: Property 'isNavOpen' does not exist on type 'App'. [plugin angular-compiler]

    src/app/app.html:10:6:
      10 │       {{ isNavOpen ? 'Close menu' : 'Open menu' }}
         ╵          ~~~~~~~~~

  Error occurs in the template of component App.

    src/app/app.ts:8:15:
      8 │   templateUrl: './app.html',
        ╵                ~~~~~~~~~~~~


✘ [ERROR] TS2339: Property 'closeNavOnMobile' does not exist on type 'App'. [plugin angular-compiler]

    src/app/app.html:19:12:
      19 │       (click)="closeNavOnMobile()"
         ╵                ~~~~~~~~~~~~~~~~

  Error occurs in the template of component App.

    src/app/app.ts:8:15:
      8 │   templateUrl: './app.html',
        ╵                ~~~~~~~~~~~~


✘ [ERROR] TS2339: Property 'isNavOpen' does not exist on type 'App'. [plugin angular-compiler]

    src/app/app.html:24:42:
      24 │     <div class="nav-collapse" [class.open]="isNavOpen">
         ╵                                             ~~~~~~~~~

  Error occurs in the template of component App.

    src/app/app.ts:8:15:
      8 │   templateUrl: './app.html',
        ╵                ~~~~~~~~~~~~


✘ [ERROR] TS2339: Property 'closeNavOnMobile' does not exist on type 'App'. [plugin angular-compiler]

    src/app/app.html:31:14:
      31 │           (click)="closeNavOnMobile()"
         ╵                    ~~~~~~~~~~~~~~~~

  Error occurs in the template of component App.

    src/app/app.ts:8:15:
      8 │   templateUrl: './app.html',
        ╵                ~~~~~~~~~~~~


✘ [ERROR] TS2339: Property 'closeNavOnMobile' does not exist on type 'App'. [plugin angular-compiler]

    src/app/app.html:43:14:
      43 │           (click)="closeNavOnMobile()"
         ╵                    ~~~~~~~~~~~~~~~~

  Error occurs in the template of component App.

    src/app/app.ts:8:15:
      8 │   templateUrl: './app.html',
        ╵                ~~~~~~~~~~~~

--- a/frontend/app/src/app/app.scss
+++ b/frontend/app/src/app/app.scss
@@ -35,6 +35,7 @@
 	font-weight: 700;
 	cursor: pointer;
 }
+
 .nav-collapse {
 	display: contents;
 }
@@ -131,12 +132,14 @@
 
 	.nav-collapse.open {
 		display: grid;
+		grid-template-columns: 1fr;
 	}
 
 	.nav-group {
 		display: grid;
 		gap: 0.5rem;
-	}	
+		align-items: stretch;
+	}
 
 	.nav-left,
 	.nav-right {

--- a/frontend/app/src/app/app.ts
+++ b/frontend/app/src/app/app.ts
@@ -9,6 +9,18 @@ import { RouterLink, RouterLinkActive, RouterOutlet } from '@angular/router';
   styleUrl: './app.scss'
 })
 export class App {
+  isNavOpen = false;
+
+  toggleNav(): void {
+    this.isNavOpen = !this.isNavOpen;
+  }
+
+  closeNavOnMobile(): void {
+    if (window.innerWidth <= 1080) {
+      this.isNavOpen = false;
+    }
+  }
+  
   readonly authenticatedScreens = [
     { label: 'Front Desk', route: '/front-desk' },
     { label: 'Race Control', route: '/race-control' },

